sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -524,10 +546,12 @@ class annotated_ref {
   public:
     annotated_ref(const annotated_ref&) = delete;
     operator T() const;
-    T operator=(const T &) const;
+    template <typename TT>
+    T operator=(T) const;
```

Comment:
```suggestion
    T operator=(T) const;
```

Reply:
That's fast LOL! Just pushed all the changes